middle_end/flambda2/to_cmm/to_cmm_static.ml

```diff
@@ -206,41 +206,41 @@ let static_const0 env res ~updates (bound_static : Bound_static.Pattern.t)
     let transl = Numeric_types.Float32_by_bit_pattern.to_float in
     let structured f = Cmmgen_state.Const_float32 f in
     let res, env, updates =
-      static_boxed_number
-        ~kind:(Single { reg = Float32 })
-        ~env ~symbol ~default ~emit:C.emit_float32_constant ~transl ~structured
-        v res updates
+      static_boxed_number ~kind:C.Update_kind.naked_float32s ~env ~symbol
+        ~default ~emit:C.emit_float32_constant ~transl ~structured v res updates
     in
     env, res, updates
   | Block_like symbol, Boxed_float v ->
     let default = Numeric_types.Float_by_bit_pattern.zero in
     let transl = Numeric_types.Float_by_bit_pattern.to_float in
     let structured f = Cmmgen_state.Const_float f in
     let res, env, updates =
-      static_boxed_number ~kind:Double ~env ~symbol ~default
+      static_boxed_number ~kind:C.Update_kind.naked_floats ~env ~symbol ~default
         ~emit:C.emit_float_constant ~transl ~structured v res updates
     in
     env, res, updates
   | Block_like symbol, Boxed_int32 v ->
     let structured i = Cmmgen_state.Const_int32 i in
     let res, env, updates =
-      static_boxed_number ~kind:Word_int ~env ~symbol ~default:0l
-        ~emit:C.emit_int32_constant ~transl:Fun.id ~structured v res updates
+      static_boxed_number ~kind:C.Update_kind.naked_int32s ~env ~symbol
```

Comment:
This means the top 32 bits of the static int32 will be zero rather than sign extended.
I believe this is fine because all int32 loads in `cmm_helpers` are sign-extending.

Reply:
I'd rather keep the corner behavior for that, if possible. Else, such a change would need a very careful review of all cmm generation code and related assumptions and comments.

Reply:
It's easy enough to keep, but surely the relevant cmm code is correct given `emit_int32_constant` already zeroes the top bits, right?

Reply:
Guillaume and I discussed.  We think it's safest to ensure that every int32 in an int64-sized slot is sign extended.  I'll push a commit soon which changes emit_int32_constant to sign extend (it's unclear why this wasn't happening before, it might very well just have been an oversight).  This particular case here can use "naked_int32_fields".

Reply:
Likewise above I'll make the equivalent change for the float32 case.

Reply:
I've also added a comment in `cmm.mli` explaning the conventions/rules.